Assets/HoloToolkit/Input/Scripts/ControllerInfo.cs

```diff
@@ -0,0 +1,227 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace HoloToolkit.Unity.InputModule
+{
+    /// <summary>
+    /// This script keeps track of the GameObjects for each button on the controller.
+    /// It also keeps track of the animation Transforms in order to properly animate according to user input.
+    /// </summary>
+    public class ControllerInfo : MonoBehaviour
+    {
+        public GameObject home;
+        public Transform homePressed;
+        public Transform homeUnpressed;
+        public GameObject menu;
+        public Transform menuPressed;
+        public Transform menuUnpressed;
+        public GameObject grasp;
+        public Transform graspPressed;
+        public Transform graspUnpressed;
+        public GameObject thumbstickPress;
+        public Transform thumbstickPressed;
+        public Transform thumbstickUnpressed;
+        public GameObject thumbstickX;
+        public Transform thumbstickXMin;
+        public Transform thumbstickXMax;
+        public GameObject thumbstickY;
+        public Transform thumbstickYMin;
+        public Transform thumbstickYMax;
+        public GameObject select;
+        public Transform selectPressed;
+        public Transform selectUnpressed;
+        public GameObject touchpadPress;
+        public Transform touchpadPressed;
+        public Transform touchpadUnpressed;
+        public GameObject touchpadPressX;
+        public Transform touchpadPressXMin;
+        public Transform touchpadPressXMax;
+        public GameObject touchpadPressY;
+        public Transform touchpadPressYMin;
+        public Transform touchpadPressYMax;
+        public GameObject touchpadTouchX;
+        public Transform touchpadTouchXMin;
+        public Transform touchpadTouchXMax;
+        public GameObject touchpadTouchY;
+        public Transform touchpadTouchYMin;
+        public Transform touchpadTouchYMax;
+        public GameObject touchpadTouchVisualizer;
+
+        // These bools and doubles are used to determine if a button's state has changed.
+        public bool wasGrasped;
+        public bool wasMenuPressed;
+        public bool wasThumbstickPressed;
+        public bool wasTouchpadPressed;
+        public bool wasTouchpadTouched;
+        public double lastThumbstickX;
+        public double lastThumbstickY;
+        public double lastTouchpadX;
+        public double lastTouchpadY;
+        public double lastSelectPressedValue;
+
+        public void LoadInfo(Transform[] childTransforms, GameObject touchpadTouchedOverride, Shader shader)
+        {
+            foreach (Transform child in childTransforms)
+            {
+                switch (child.name.ToLower())
```

Comment:
I added a summary comment at the top to describe what it's doing. Let me know if you think comments along the way are still needed.